src/black/__init__.py

```diff
@@ -741,6 +765,12 @@ def format_file_in_place(
         dst_contents = format_file_contents(src_contents, fast=fast, mode=mode)
     except NothingChanged:
         return False
+    except ModuleNotFoundError:
+        warnings.warn(
+            f"Skipping '{src}' as Jupyter dependencies are not installed.\n"
+            "You can fix this by running ``pip install black[jupyter]``"
+        )
+        return False
```

Comment:
I don't like very much since it's kinda tacked on and could easily become misleading if we ever add additional lazy / conditional imports. It's also not immediately clear that this situation must be about missing Jupyter dependencies.

Reply:
This is also what's causing the caching bug I noted above.

Reply:
Excellent points here, thanks for catching this!